### DISCUSSION TITLE
Microck gPRC mock, GraphQLController always returns HttpStatus.OK

### DISCUSSION THREAD

**Discussion by sgpranith** [microcks/microcks#1048] [Q&A]
Hello Microck Team,

In the Microck gPRC mock, the GraphQLController always returns HttpStatus.OK instead of actually configuring status (see https://github.com/microcks/microcks/blob/master/webapp/src/main/java/io/github/microcks/web/GraphQLController.java#L261). When compared to the RestController (referenced in https://github.com/microcks/microcks/blob/master/webapp/src/main/java/io/github/microcks/web/RestController.java#L261), the actual status in the mock response is returned. I am trying to understand if there is any particular reason why the GraphQLController only returns HttpStatus.OK.

**Comment by lbroudoux** [comment]
Hi @sgpranith,

To my knowledge and understanding of GraphQL, most libraries only return one of two codes -- 400 if the request itself is somehow invalid and 200 otherwise. I think IIRC that this is what we implemented for now in the GraphQL controller. Maybe things have changed a bit or need to be refined if we can now specify some other return codes using a GraphQL schema. Let us know what you have in mind.
